game-core/src/main/java/games/strategy/engine/data/PlayerID.java

```diff
@@ -120,27 +127,37 @@ public String getType() {
   }
 
   /**
-   * First string is "Human" or "AI", while second string is the name of the player. Separated with a colon.
+   * First string is "Human" or "AI" or "null" (case insensitive), while second string is the name of the player,
+   * separated with a colon. For example, it could be "AI:Hard (AI)".
+   *
+   * @throws IllegalArgumentException If {@code encodedType} does not contain two strings separated by a colon; or if
+   *         the first string is not one of "AI", "Human", or "null" (case insensitive).
    */
-  public void setWhoAmI(final String encodedPlayerTypeAndName) {
-    // so for example, it should be "AI:Hard (AI)"
-    final String[] s = encodedPlayerTypeAndName.split(":");
-    if (s.length != 2) {
-      throw new IllegalStateException(String.format("whoAmI '%s' must have two strings, separated by a colon",
-          encodedPlayerTypeAndName));
-    }
-    if (!(s[0].equalsIgnoreCase("AI") || s[0].equalsIgnoreCase("Human") || s[0].equalsIgnoreCase("null"))) {
-      throw new IllegalStateException("whoAmI first part must be, ai or human or client");
-    }
-    m_whoAmI = encodedPlayerTypeAndName;
+  public void setWhoAmI(final String encodedType) {
+    final List<String> tokens = tokenizeEncodedType(encodedType);
+    checkArgument(tokens.size() == 2, "whoAmI '" + encodedType + "' must have two strings, separated by a colon");
+    final String typeId = tokens.get(0);
+    checkArgument(
+        "AI".equalsIgnoreCase(typeId) || "Human".equalsIgnoreCase(typeId) || "null".equalsIgnoreCase(typeId),
+        "whoAmI '" + encodedType + "' first part must be, ai or human or null");
+    m_whoAmI = encodedType;
+  }
+
+  private static List<String> tokenizeEncodedType(final String encodedType) {
+    return Splitter.on(':').splitToList(encodedType);
   }
 
   public String getWhoAmI() {
     return m_whoAmI;
   }
 
+  public Type getPlayerType() {
```

Comment:
I couldn't name this method `getType()` due to a collision with the method of that name inherited from `NamedUnitHolder`.